### ISSUE TITLE
[Bug]: Height not recalculated when task/permission is denied

### ISSUE THREAD

**Issue by srothgan** [srothgan/claude-code-rust#12]
### Description

When a permission prompt (tool call) is denied by the user, the block visually shrinks (the permission UI disappears) but the cached height measurement is not invalidated. This causes the layout to use stale height values, leading to incorrect scroll positions and visual gaps.

### Steps to Reproduce

1. Start a session and trigger a tool call that requires permission
2. Deny the permission prompt
3. Observe the block height stays at the old (larger) value in the layout

### Expected Behavior

Denying a permission prompt should invalidate the block's cached height so the layout reflows correctly.

### Actual Behavior

The block cache / height measurement does not get triggered on denial, leaving stale height values.

### Version

claude-rust 0.1.3

### Operating System

Windows (x86_64)

### Rust Version

rustc 1.93.1 (01f6ddf75 2026-02-11)

### Relevant Log Output

```shell

```

### Implementation Notes (optional)

- The `BlockCache` stores height per width -- denial needs to invalidate the entry for the affected tool call
- Look at where permission denial is handled and ensure it calls the same cache invalidation path as other block content changes

### Checklist

- [x] I have searched existing issues to ensure this is not a duplicate
